### ISSUE TITLE
v8.9 branch seems to depend on code added after v8.9?

### ISSUE THREAD

**Issue by Ptival** [uwplse/coq-change-analytics#48]
In particular, `Stm.document_add_hook`:

https://github.com/uwplse/coq-change-analytics/blob/55dd66d7aa7a986b7fbb568562065325f5b00e5b/plugin/src/analytical.ml#L548

seems to be missing in coq-v8.9.1

**Comment by tlringer** [comment]
You need a custom version of Coq 8.9, which is in the 8.9 branch. The 8.9 branch should pull that version for you when you run ./build.sh if you tell it you have not already done so. We had to backport the hook to a branch of 8.9 (similarly 8.8) so you unfortunately can't use standard Coq if you need 8.9 or 8.8.

Please let me know if this works for you.

**Comment by Ptival** [comment]
Oh I see. Sorry, I was trying to build it with nix and circumvented your whole `build.sh`.

I am now in the process of building it, if this helps anyone else, this is a nix shell expression that installs the proper "patched" version of 8.9:

```
{ nixpkgs ? import <nixpkgs> {}
}:
with nixpkgs;
stdenv.mkDerivation {
  buildInputs =
    (with ocamlPackages; [
      camlp5
      cohttp
      cohttp-lwt-unix
      findlib
      ocaml
      sexplib
    ])
    ++
    [
      (coq_8_9.overrideAttrs(oldAttrs: {
        src = fetchFromGitHub {
          owner = "tlringer";
          repo = "coq";
          rev = "v8.9-backport";
          sha256 = "110rzcbp35ayy0fl3k0sn2imwh6alwp837pwj0w8plsawgb42zws";
        };
      }))
    ]
    ;
  name = "coq-change-analytics";
}
```

It is easy to adapt to 8.8 too. Sadly, 8.10 is currently not building for nix, due to GTK3 issues, and people interested may find out more at:
https://github.com/NixOS/nixpkgs/issues/66217
